### ISSUE TITLE
Question regarding the "baopenr" module

### ISSUE THREAD

**Issue by cfbastarz** [NOAA-EMC/NCEPLIBS-nemsio#64]
Hello,

I'm using the `nemsio` lib (as a matter of fact, an older version tagged as 2.2.2 - I could not find this version at the releases page in this repository) and got stuck with an error at:

https://github.com/NOAA-EMC/NCEPLIBS-nemsio/blob/48bda6fa65a2564f237eaf56c9af1ddfd4d19f42/src/nemsio_openclose.f90#L495

My code is being compiled with the Intel compiler (the following modules are loaded in a Dell cluster):

```
Currently Loaded Modules:
1) autotools      6) compiler-rt/2021.4.0  11) mkl/2021.4.0    16) netcdf/4.7.4
2) prun/2.2       7) debugger/2021.6.0     12) intel/2022.1.0  17) netcdf-fortran/4.5.3
3) ohpc           8) dpl/2021.4.0          13) mpi/2021.4.0
4) hwloc/2.5.0    9) oclfpga/2021.4.0      14) impi/2021.4.0
5) tbb/2021.4.0  10) compiler/2022.1.0     15) phdf5/1.10.8
```

The version of the Intel compiler I'm using is:

```
mpif90 --version
ifort (IFORT) 2021.6.0 20220226
Copyright (C) 1985-2022 Intel Corporation.  All rights reserved.
```

Including debug flags to compile the `nemsio` lib (`-O0 -g -convert big_endian -traceback -check all -debug all -ftrapuv -fpe0-fstack-protector-all -warn all -fPIC`), I get the following error at the runtime:

```
Boundary Run-Time Check Failure for variable 'nemsio_openclose_mp_nemsio_open_$IOS'
```
Digging into the code, I can see that the referred `ios` is an I/O return code provided by the mentioned `baopenr` module, which I guess is from the `bacio` library (is that correct?).

On the other hand, when compiling with the GNU compiler environment I get no errors.

Since I'm not using a supported version of `nemsio` (it seems that the version I'm using was modified/moded at some point), any advice on how to tackle this problem is greatly appreciated. 

Thank you,

Carlos

**Comment by edwardhartnett** [comment]
Can you try a released version of nemsio?

**Comment by cfbastarz** [comment]
@edwardhartnett, yes I can. I just tried versions 2.4.0 and 2.5.4 (the first and the last releases) and all of them yelds the same result. In the `nemsio` release I'm using (sorry, I don't know where it came from), I see that there is an extra source file called `nemsio_gfs.f90`, which does not integrate any of those versions. I wonder if it was a part of some `nemsio` release at some point or so. Anyways, in my tests I just kept this file and replaced the remainder with the ones from the mentioned versions. I've also tried some different debug flags and optimization levels, all to no avail. The code compiles, but it chokes. I guess it might be something with the version of the Intel compiler I'm using or even the modules I'm using.

Thank you.

**Comment by edwardhartnett** [comment]
OK, let's concern ourselves with only the most recent release.

The release does not build for you? Or you have some code that targets the release that does not work for you?

If we think there is a bug we need to construct a test case that demonstrates the bug...

**Comment by cfbastarz** [comment]
@edwardhartnett, I've sorted it out. I was mixing the `bacio` and `w3lib` from my source code with the most recent release of `nemsio`. The compilation seems to be fine, but the code chokes when it runs. I've downloaded the `NCEPLIBS-[bacio,w3emc,nemsio]` injected the `nemsio_gfs.f90` that I need into the `nemsio` release (I've changed the `CMakeLists.txt` accordingly) and compiled all the libs using the provided `cmake` structure (my code just use a bunch of Makefiles). After that, I hooked the newly compiled libs into my Makefiles and compiled the rest of my code. It worked fine and there is no problem anymore.

Thank you!
